### DISCUSSION TITLE
Computing energy consumption from power measurements

### DISCUSSION THREAD

**Discussion by ejhusom** [hubblo-org/scaphandre#375] [Q&A]
Hi,

I'm using Scaphandre to monitor the power usage of a certain process. My goal is to simply **compute the energy consumption** (in Joules) for this process running for a given duration. However, I'm not sure if I fully understand the power measurements coming from Scaphandre. Are they the current power level at the time of measuring? So in order to compute the energy consumption $E$ over a duration $t$ with $N$ number of power measurements $P_n$, I would have to compute the average power level $\hat{P}$ and multiply by the duration $t$:

$$E =\frac{t}{N} \sum_{n=0}^N P_n$$

Or in Python:

```python
energy_consumption = duration * power_measurements.sum() / len(power_measurements)
```

This would then be independent of the time between measurements, but the shorter time I have, the more accurate the readings will be?

I apologize for asking very basic questions, but I would like to confirm that this is correct :)

**Comment by JJ** [comment]
Well, depends on sampling time; the more sampling time is adjusted to your running time with no left-over the more accurate it will be.
